gpkit/constraints/bounded.py

```diff
@@ -25,12 +24,14 @@ def varkey_bounds(varkeys, lower, upper):
     constraints = []
     for varkey in varkeys:
         variable = Variable(**varkey.descr)
-        units = varkey.units if isinstance(varkey.units, Quantity) else 1
+        if variable.units:
+            variable.hmap.units = None
+            variable.units = None
```

Comment:
No, I'm convinced there really is something funny with hashes going on.

Here's a simple example so that we can be concrete:

```python
from gpkit import Variable, Model


x = Variable("x", "ft")
y = Variable("y", 3, "ft")

xbound = Variable(**x.key.descr)
xbound.hmap.units = None
xbound.units = None

m = Model(x, [x >= y,
              xbound >= 100])
sol = m.solve()
print sol.table()
```

Running this, the optimal cost is 100 and so `xbound` does have the desired behavior of bounding `x`, so I agree with you that this works. But I really, really don't like WHY it works.

```python
In [4]: hash(x.key)
Out[4]: -6658453509673149733

In [5]: hash(xbound.key)
Out[5]: -6658453509673149733

In [6]: x2 = Variable("x")

In [7]: hash(x2.key)
Out[7]: 4376088834770714827

In [8]: x2 = Variable("x", "ft")

In [9]: hash(x2.key)
Out[9]: -6658453509673149733
```

See, `Variable("x").key` has a different hash than `Variable("x", "ft").key`. But `xbound.key` has the SAME hash as `Variable("x", "ft").key`.

As I understand it, this works because you leave the units of the key intact (`xbound.key.units` is `ft`), but set the units of the `Variable` to `None`.

It doesn't bother you to have a bunch of Variables floating around whose keys have different units than their Variable? What if I for whatever reason reimplement  `Variable.units` to simply return `Variable.key.units`?